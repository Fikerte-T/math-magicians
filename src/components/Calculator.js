import React, { useState } from 'react';
import calculate from '../logic/calculate.js';

const Calculator = () => {
  const [state, setState] = useState({ total: '', next: '', operation: '' });
  const updateStates = (newResult) => {
    setState((prevState) => ({
      ...prevState, total: newResult.total, next: newResult.next, operation: newResult.operation,
    }));
  };
  const { next, total, operation } = state;
  let result = '0';
  if (total) {
    result = `${total} ${operation || ''} ${next || ''}`;
  } else if (next) {
    result = next;
  }

  const clickHandler = (state, bName) => {
    const obj = calculate(state, bName);
    updateStates(obj);
  };
  return (
          <div className="calc-page">
              <h3 className="calc-title">Let's do some Math!</h3>
                <div className="wrapper">
                  <div className="result">{result || '0'}</div>
                  <div className="buttons-container">
                      <button name="AC" onClick = {(e) => clickHandler(state, e.target.name)} className="button">AC</button>
                      <button name="+/-" onClick = {(e) => clickHandler(state, e.target.name)} className="button">+/-</button>
                      <button name="%" onClick = {(e) => clickHandler(state, e.target.name)} className="button">%</button>
                      <button name="รท" onClick = {(e) => clickHandler(state, e.target.name)} className="button sign">รท</button>
                      <button name="7" onClick = {(e) => clickHandler(state, e.target.name)} className="button">7</button>
                      <button name="8" onClick = {(e) => clickHandler(state, e.target.name)} className="button">8</button>
                      <button name="9" onClick = {(e) => clickHandler(state, e.target.name)} className="button">9</button>
                      <button name="x" onClick = {(e) => clickHandler(state, e.target.name)} className="button sign">x</button>
                      <button name="4" onClick = {(e) => clickHandler(state, e.target.name)} className="button">4</button>
                      <button name="5" onClick = {(e) => clickHandler(state, e.target.name)} className="button">5</button>
                      <button name="6" onClick = {(e) => clickHandler(state, e.target.name)} className="button">6</button>
                      <button name="-" onClick = {(e) => clickHandler(state, e.target.name)} className="button sign">-</button>
                      <button name="1" onClick = {(e) => clickHandler(state, e.target.name)} className="button">1</button>
                      <button name="2" onClick = {(e) => clickHandler(state, e.target.name)} className="button">2</button>
                      <button name="3" onClick = {(e) => clickHandler(state, e.target.name)} className="button">3</button>
                      <button name="+" onClick = {(e) => clickHandler(state, e.target.name)} className="button sign">+</button>
                      <button name="0" onClick = {(e) => clickHandler(state, e.target.name)} className="button zero">0</button>
                      <button name="." onClick = {(e) => clickHandler(state, e.target.name)} className="button">.</button>
                      <button name="=" onClick = {(e) => clickHandler(state, e.target.name)} className="button">=</button>
                  </div>
                </div>
          </div>
  );
};

export default Calculator;
