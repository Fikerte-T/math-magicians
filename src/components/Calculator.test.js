import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator.js';

it('calculator snapshot test', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});