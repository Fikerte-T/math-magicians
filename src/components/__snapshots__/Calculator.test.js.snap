// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator snapshot test 1`] = `
<div
  className="page calc-page"
>
  <h2
    className="calc-title"
  >
    Let's do some Math!
  </h2>
  <div
    className="wrapper"
  >
    <div
      className="result"
    >
      0
    </div>
    <div
      className="buttons-container"
    >
      <button
        className="button"
        name="AC"
        onClick={[Function]}
      >
        AC
      </button>
      <button
        className="button"
        name="+/-"
        onClick={[Function]}
      >
        +/-
      </button>
      <button
        className="button"
        name="%"
        onClick={[Function]}
      >
        %
      </button>
      <button
        className="button sign"
        name="รท"
        onClick={[Function]}
      >
        รท
      </button>
      <button
        className="button"
        name="7"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="button"
        name="8"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="button"
        name="9"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="button sign"
        name="x"
        onClick={[Function]}
      >
        x
      </button>
      <button
        className="button"
        name="4"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="button"
        name="5"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="button"
        name="6"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="button sign"
        name="-"
        onClick={[Function]}
      >
        -
      </button>
      <button
        className="button"
        name="1"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="button"
        name="2"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="button"
        name="3"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="button sign"
        name="+"
        onClick={[Function]}
      >
        +
      </button>
      <button
        className="button zero"
        name="0"
        onClick={[Function]}
      >
        0
      </button>
      <button
        className="button"
        name="."
        onClick={[Function]}
      >
        .
      </button>
      <button
        className="button"
        name="="
        onClick={[Function]}
      >
        =
      </button>
    </div>
  </div>
</div>
`;
